platforms/documentation/docs/src/docs/userguide/best-practices/best_practices_general.adoc

```diff
@@ -175,3 +175,199 @@ include::sample[dir="snippets/bestPractices/avoidInternal/groovy",files="build.g
 === Tags
 
 <<tags_reference.adoc#tag:upgrades,`#upgrades`>>
+
+[[modularize_builds]]
+== Modularize Your Builds
+:keywords: structuring-builds
+
+Modularize your builds by creating subprojects.
+
+=== Explanation
+
+Splitting your build's source into multiple modules (Gradle projects) is key to taking advantage of Gradle's automatic work avoidance and parallelization features.
+After making a change to a source file, Gradle only needs to recompile affected projects.
+If all your source lives in a single project, Gradle can't skip any recompilations and won't be able to run tasks in parallel.
+You'll likely find other performance benefits of splitting source between projects due to minimizing the compilation classpath of each subproject to only the required dependencies, only running annotation processors on the source files that need them, and so on.
+
+It makes sense to split your build's source into multiple projects _early_.
+Don't wait until you hit some arbitrary number of source files or classes to do this, instead structure your build into multiple projects from the start using whatever natural boundaries exist in your codebase.
+This helps you plan for future growth as your build matures, and improves comprehensibility as each project can be described independently.
```

Comment:
I would remove this sentence. I feel like the previous statement is stronger without this additional sentence:
```suggestion
```